draft-ietf-tls-esni.md

```diff
@@ -425,16 +412,27 @@ The client-facing server computes ClientHelloInner by reversing this process.
 It scans EncodedClientHelloInner for an "outer_extensions" extension. If there
 is none, the ClientHelloInner is the EncodedClientHelloInner. Otherwise, it
 replaces the extension with the corresponding sequence of extensions in
-the ClientHelloOuter. If any referenced extensions are missing, the server
-MUST abort the connection with an "illegal_parameter" alert.
-The server then computes the digest of the reconstructed
-ClientHelloInner. If the digest does not equal OuterExtensions.inner_digest,
-then the server MUST abort the connection with a "decrypt_error" alert.
+the ClientHelloOuter. If any referenced extensions are missing or if
+"encrypted_client_hello" appears in the list, the server MUST abort the
+connection with an "illegal_parameter" alert.
 
 The "outer_extensions" extension is only used for compressing the
 ClientHelloInner. It MUST NOT be sent in either ClientHelloOuter or
 ClientHelloInner.
 
+## Authenticating the Outer ClientHello {#authenticating-outer-clienthello}
+
+To prevent a network attacker from modifying the reconstructed ClientHelloInner
+(see {{flow-clienthello-malleability}}), ECH authenticates ClientHelloOuter by
+deriving a ClientHelloOuterAAD value. This is computed by removing the
+"encrypted_client_hello" extension from ClientHelloOuter and serializing it.
+ClientHelloOuterAAD is then passed as the associated data parameter to the
+encryption.
```

Comment:
`ClientHelloOuterAAD` is a bit different than we discussed in #323. In particular, I thought it was decided that we should predict the length of the "encrypted_client_hello" extension and authenticate as many zeros. What changed your mind? For the record, the approach is here seems a lot simpler to me :)

Reply:
Oh I probably was just unclear in that discussion. I never preferred the zeros formulation. :-) That was more me musing on how annoying it would be if there were another reason to prefer it. I like this better because, as you say, it's simpler.